tests/cpp/test_argsort_device_func.cpp

```diff
@@ -0,0 +1,336 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+#include <gtest/gtest.h>
+
+#include <tests/cpp/argsort_test_helper.h>
+#include <tests/cpp/utils.h>
+#include <tests/cpp/validator.h>
+
+#include <ATen/cuda/CUDAContext.h>
+#include <cstdint>
+#include <vector>
+
+namespace nvfuser {
+
+using ArgSortDeviceFuncTest = NVFuserTest;
+
+// Parameterized test fixture for comprehensive validation
+using ArgSortComprehensiveTest = NVFuserFixtureParamTest<std::pair<int, int>>;
+
+template <typename DataT>
+std::vector<DataT> getVector(at::Tensor tensor) {
+  NVF_ERROR_EQ(tensor.dim(), 1);
+  if (tensor.dtype() == at::kBFloat16) {
+    tensor = tensor.to(at::kFloat);
+  }
+  auto cpu_tensor = tensor.cpu();
+  auto total_elements = tensor.size(0);
+  return std::vector<DataT>(
+      cpu_tensor.data_ptr<DataT>(),
+      cpu_tensor.data_ptr<DataT>() + total_elements);
+}
+
+// Helper function to validate sorting correctness
+template <typename DataT>
+bool validateArgsortOrder(
```

Comment:
Is the reason here because we are not doing stable sort?! So we can't really use a reference implementation to check the result.

Reply:
This felt quite verbose BTW.

a nitpick is that we can use aten API to check the end result:
```
>>> x = torch.randn(1024)
>>> ind = x.argsort()

>>> x[ind]
>>> sorted_x = x[ind]
>>> assert((sorted_x[0:-1] <= sorted_x[1:]).min())
```

Reply:
You're right. Let me clean it up in the next PR.